c2rust-transpile/tests/snapshots/snapshots__transpile@arrays.c.snap

```diff
@@ -0,0 +1,408 @@
+    i = i + 1;
+    *buffer
+        .offset(
+            fresh40 as isize,
+        ) = simple[7 as std::ffi::c_int as usize] as std::ffi::c_int;
+    let fresh41 = i;
+    i = i + 1;
+    *buffer
+        .offset(
+            fresh41 as isize,
+        ) = *foo.offset(0 as std::ffi::c_int as isize) as std::ffi::c_int;
+    let fresh42 = i;
+    i = i + 1;
+    *buffer
+        .offset(
+            fresh42 as isize,
+        ) = *foo.offset(1 as std::ffi::c_int as isize) as std::ffi::c_int;
+    let fresh43 = i;
+    i = i + 1;
+    *buffer
+        .offset(
+            fresh43 as isize,
+        ) = *foo.offset(2 as std::ffi::c_int as isize) as std::ffi::c_int;
+    let fresh44 = i;
+    i = i + 1;
+    *buffer
+        .offset(
+            fresh44 as isize,
+        ) = *foo.offset(3 as std::ffi::c_int as isize) as std::ffi::c_int;
+    let fresh45 = i;
+    i = i + 1;
+    *buffer
+        .offset(
+            fresh45 as isize,
+        ) = *foo.offset(4 as std::ffi::c_int as isize) as std::ffi::c_int;
+    let fresh46 = i;
+    i = i + 1;
+    *buffer
+        .offset(
+            fresh46 as isize,
+        ) = *foo.offset(5 as std::ffi::c_int as isize) as std::ffi::c_int;
+    let fresh47 = i;
+    i = i + 1;
+    *buffer
+        .offset(
+            fresh47 as isize,
+        ) = *foo.offset(6 as std::ffi::c_int as isize) as std::ffi::c_int;
+    let fresh48 = i;
+    i = i + 1;
+    *buffer
+        .offset(
+            fresh48 as isize,
+        ) = *foo.offset(7 as std::ffi::c_int as isize) as std::ffi::c_int;
+    let mut past_end: *mut std::ffi::c_char = &mut *simple
+        .as_mut_ptr()
+        .offset(
+            ::core::mem::size_of::<[std::ffi::c_char; 9]>() as std::ffi::c_ulong as isize,
+        ) as *mut std::ffi::c_char;
+    past_end = &mut *foo.offset(8 as std::ffi::c_int as isize) as *mut std::ffi::c_char;
+}
+#[no_mangle]
+pub unsafe extern "C" fn short_initializer() {
+    let mut empty_brackets: [std::ffi::c_int; 16] = [0; 16];
+    let mut brackets_with_zero: [std::ffi::c_int; 16] = [0 as std::ffi::c_int; 16];
+    let mut brackets_with_one: [std::ffi::c_int; 4] = [1 as std::ffi::c_int, 0, 0, 0];
+    let mut excess_elements_1: [std::ffi::c_int; 2] = [
+        1 as std::ffi::c_int,
+        2 as std::ffi::c_int,
+    ];
+    let mut excess_elements_2: [std::ffi::c_int; 0] = [0; 0];
+    let mut single_struct: [C2RustUnnamed_2; 1] = [
+        {
+            let mut init = C2RustUnnamed_2 {
+                x: 1 as std::ffi::c_int as std::ffi::c_short,
+                y: 2 as std::ffi::c_int,
+            };
+            init
+        },
+    ];
+    let mut many_struct: [C2RustUnnamed_1; 3] = [
+        {
+            let mut init = C2RustUnnamed_1 {
+                x: 1 as std::ffi::c_int as std::ffi::c_short,
+                y: 2 as std::ffi::c_int,
+            };
+            init
+        },
+        C2RustUnnamed_1 { x: 0, y: 0 },
+        C2RustUnnamed_1 { x: 0, y: 0 },
+    ];
```

Comment:
in more recent versions this could be e.g.

```rust
let mut many_struct: [C2RustUnnamed_1; 3] = const { 
    let mut base = [C2RustUnnamed_1 { x: 0, y: 0 }; 3];
    base[0] = C2RustUnnamed_1 {
        x: 1 as std::ffi::c_int as std::ffi::c_short,
        y: 2 as std::ffi::c_int,
    };

    base
};
```

but idk, it's extra work and may not actually come up in practice.

Reply:
Yeah, this is definitely cleaner.  How recently was support for this added, though?  I know the `const {}` blocks are new, but you don't need that for this, right?

Reply:
True, you just have to name the `const` item, but I guess if you nest it in a block the name can never conflict?